lib/nerves/utils/wsl.ex

```diff
@@ -26,7 +26,12 @@ defmodule Nerves.Utils.WSL do
   def running_on_wsl?(osrelease_path \\ "/proc/sys/kernel/osrelease") do
     with true <- File.exists?(osrelease_path),
          {content, _} <- Nerves.Port.cmd("cat", [osrelease_path]) do
-      Regex.match?(~r/[Mm]icrosoft/, content)
+      # Docker Desktop for Windows uses WSL 2 kernel as the back-end
+      # This line checks whether the env is Docker or WSL 2 shell
+      case !File.exists?("/.dockerenv") do
+        true -> Regex.match?(~r/[Mm]icrosoft/, content)
+        false -> false
+      end
     else
       _ ->
         false
```

Comment:
It looks like this code could be simplified. Would something like this work?

```elixir
  def running_on_wsl?() do
    # Docker Desktop for Windows uses WSL 2 kernel as the back-end
    # so also check whether the env is not Docker
    Regex.match?(~r/[Mm]icrosoft/, osrelease()) and not File.exists?("/.dockerenv")
  end

  defp osrelease() do
    case File.read("/proc/sys/kernel/osrelease") do
      {:ok, text} -> text
      _ -> "unknown"
  end
```

Reply:
Thanks! This code is smarter than mine!!
Obviously there is no problem at all, but I think we need to make sure this works properly. I will be able to check it early next week. please wait a moment🙏